File: readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

float findLetters(string s);
float findSenteces(string s);
float theEqation(int L , int S);
void control(float inx);
int main(void)
{
    string inputString = get_string("Text: ");
    int L = findLetters(inputString);
    int S = findSenteces(inputString);
    float index = theEqation(L , S);

    control(index);


}
float findLetters(string s)
{
    float letters = 0 ;
    float words = 0;
    int i = 0 ;
    while(s[i] != '\0')
    {
        if(isalpha(s[i]))
            letters++;
        if (isblank(s[i]))
             words++;
        i++;
    }
    words++;
    //printf("letters  %0.2f\n",letters);
   // printf("wordsL  %0.2f\n",words);
    return (letters/words)*100.0;
}

float findSenteces(string s)
{
    float sentences = 0 ;
    float words = 0 ;
    int i = 0 ;
    while(s[i] != '\0')
    {
        if( s[i] == '.' || s[i] == '!' || s[i] == '?')
            sentences++;
        else if (s[i] == ' ')
             words++;
        i++;
    }
    words++;
    //printf("sentences  %0.2f\n",sentences);
    //printf("words  %0.2f\n",words);
    return (sentences/words)*100.0;
}

float theEqation(int L , int S)
{
    float index = 0.0588 * L - 0.296 * S - 15.8;
    return index;
}
void control(float inx)
{
    if (inx < 1)
        printf("Before Grade 1\n");
    else if (inx >= 1 && inx <2)
        printf("Grade 1\n");
    else if (inx >= 2 && inx <3)
        printf("Grade 2\n");
    else if (inx >= 3 && inx <4)
        printf("Grade 3\n");
    else if (inx >= 4 && inx <5)
        printf("Grade 4\n");
    else if (inx >= 5 && inx <6)
        printf("Grade 5\n");
    else if (inx >= 6 && inx <7)
        printf("Grade 6\n");
    else if (inx >= 7 && inx <8)
        printf("Grade 7\n");
    else if (inx >= 8 && inx <9)
        printf("Grade 8\n");
    else if (inx >= 9 && inx <10)
        printf("Grade 9\n");
    else if (inx >= 10 && inx <11)
        printf("Grade 10\n");
    else if (inx >= 11 && inx <12)
        printf("Grade 11\n");
     else if (inx >= 12 && inx <13)
        printf("Grade 12\n");
    else if (inx >= 13 && inx <14)
        printf("Grade 13\n");
    else if (inx >= 14 && inx <15)
        printf("Grade 14\n");
    else if (inx >= 15 && inx <16)
        printf("Grade 15\n");
    else if (inx >= 16 && inx <17)
        printf("Grade 16\n");
    else
        printf("Grade 16+\n");

}

File: sentimental-readability/readability.py
import re

def main():

    text = input ("Text: ")
    letters = countLetters(text)

    words = countWords(text)
    sentences = countSentences(text)
    Lv = (letters / words) * 100
    Sv = (sentences / words) * 100
    theValue = calculateTheValue(Lv , Sv)

    findResult(theValue)

def countLetters(text):
     letters = len([char for char in text if char.isalpha()])
     return letters


def countWords(text):
    words = len(text.split())
    return words


def countSentences(text):
    sentences = len(re.split(r'[.!?]+', text))
    return sentences -1


def calculateTheValue(L,S):
    index = 0.0588 * L - 0.296 * S - 15.8
    return index



def findResult(inx):
    if inx < 1:
        print("Before Grade 1")
    elif 1 <= inx < 2:
        print("Grade 2")
    elif 2 <= inx < 3:
        print("Grade 3")
    elif 3 <= inx < 4:
        print("Grade 4")
    elif 4 <= inx < 5:
        print("Grade 5")
    elif 5 <= inx < 6:
        print("Grade 6")
    elif 6 <= inx < 7:
        print("Grade 7")
    elif 7 <= inx < 8:
        print("Grade 8")
    elif 8 <= inx < 9:
        print("Grade 9")
    elif 9 <= inx < 10:
        print("Grade 10")
    elif 10 <= inx < 12:
        print("Grade 10")
    elif 11 <= inx < 13:
        print("Grade 11")
    elif 12 <= inx < 14:
        print("Grade 12")
    elif 13 <= inx < 15:
        print("Grade 13")
    elif 14 <= inx < 16:
        print("Grade 14")
    elif 15 <= inx < 16:
        print("Grade 15")
    elif 16 <= inx < 17:
        print("Grade 16")
    else:
        print("Grade 16+")


main()
